protocol/scripts/iaviewer/unmarshaller.go

```diff
@@ -0,0 +1,32 @@
+package main
+
+import (
+	"reflect"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	app "github.com/dydxprotocol/v4-chain/protocol/app"
+	clob "github.com/dydxprotocol/v4-chain/protocol/x/clob/types"
+)
+
+func protoUnmarshaller[M codec.ProtoMarshaler](b []byte) string {
+	cdc := app.GetEncodingConfig().Codec
+	var m M
+	// TODO: avoid reflection?
+	msgType := reflect.TypeOf(m).Elem()
+	m = reflect.New(msgType).Interface().(M)
+	cdc.MustUnmarshal(b, m)
+	return m.String()
```

Comment:
Q: should we return string here and use %s? or should we return `m` and use %+v?

not sure if %+v works

Reply:
The functions would have generic return value which results in different return values for each proto. We need all the return values to be the same so we can put them in a map like we're doing now.

Reply:
you can use `any`? 

not a big deal though

Reply:
Then to use `String()` I'd need to type assert which would require another generic function. (or maybe reflection works?)

I don't know how %+v works, but I don't think it would work on `interface{}`.